src/preferences/broadcastprofile.cpp

```diff
@@ -0,0 +1,466 @@
+const char* kReconnectPeriod = "ReconnectPeriod";
+const char* kServertype = "Servertype";
+const char* kStreamDesc = "StreamDesc";
+const char* kStreamGenre = "StreamGenre";
+const char* kStreamName = "StreamName";
+const char* kStreamPublic = "StreamPublic";
+const char* kStreamWebsite = "StreamWebsite";
+
+const double kDefaultBitrate = 128;
+const int kDefaultChannels = 2;
+const bool kDefaultEnableMetadata = false;
+const bool kDefaultEnableReconnect = true;
+const bool kDefaultLimitReconnects = true;
+const int kDefaultMaximumRetries = 10;
+// No tr() here, see https://bugs.launchpad.net/mixxx/+bug/1419500
+const QString kDefaultMetadataFormat("$artist - $title");
+const bool kDefaultNoDelayFirstReconnect = true;
+const bool kDefaultOggDynamicupdate = false;
+double kDefaultReconnectFirstDelay = 0.0;
+double kDefaultReconnectPeriod = 5.0;
+const QString kDefaultStreamDesc =
+        QObject::tr("This stream is online for testing purposes!");
+const QString kDefaultStreamGenre = QObject::tr("Live Mix");
+const bool kDefaultStreamPublic = false;
+
+const QRegExp kForbiddenChars = QRegExp("[<>:\"\\/|?*\\\\]");
+} // anonymous namespace
+
+BroadcastProfile::BroadcastProfile(const QString& profileName) {
+    setDefaultValues();
+    setProfileName(profileName);
+}
+
+bool BroadcastProfile::validName(const QString& str) {
+    return !str.contains(kForbiddenChars);
+}
+
+BroadcastProfile* BroadcastProfile::loadFromFile(const QString& filename) {
+    QFileInfo xmlFile(filename);
+    if(!xmlFile.exists())
+        return nullptr;
+
+    QString profileName = xmlFile.baseName();
+
+    BroadcastProfile* profile = new BroadcastProfile(profileName);
+    profile->loadValues(filename);
+
+    return profile;
+}
+
+void BroadcastProfile::setDefaultValues() {
+    m_enabled = true;
+
+    m_host = QString();
+    m_port = BROADCAST_DEFAULT_PORT;
+    m_serverType = QString();
+    m_login = QString();
+    m_password = QString();
+
+    m_enableReconnect = kDefaultEnableReconnect;
+    m_reconnectPeriod = kDefaultReconnectPeriod;
+    m_limitReconnects = kDefaultLimitReconnects;
+
+    m_mountpoint = QString();
+    m_streamDesc = kDefaultStreamDesc;
+    m_streamGenre = kDefaultStreamGenre;
+    m_streamName = QString();
+    m_streamPublic = kDefaultStreamPublic;
+    m_streamWebsite = MIXXX_WEBSITE_URL;
+
+    m_enableMetadata = kDefaultEnableMetadata;
+    m_metadataCharset = QString();
+    m_customArtist = QString();
+    m_customTitle = QString();
+    m_metadataFormat = kDefaultMetadataFormat;
+    m_oggDynamicUpdate = kDefaultOggDynamicupdate;
+
+    m_bitrate = kDefaultBitrate;
+    m_channels = kDefaultChannels;
+    m_format = QString();
+
+    m_noDelayFirstReconnect = kDefaultNoDelayFirstReconnect;
+    m_reconnectFirstDelay = kDefaultReconnectFirstDelay;
+    m_maximumRetries = kDefaultMaximumRetries;
+}
+
+void BroadcastProfile::loadValues(const QString& filename) {
+    QDomElement doc = XmlParse::openXMLFile(filename, kDoctype);
```

Comment:
A call to `XmlParse::openXMLFile` on a non-existing document pops up an error message ("There was a problem parsing the XML file [...]") and returns an empty `QDomElement`. In my current codebase I added a children count check after that call to avoid writing to attributes if the element has no children.